numpy/random/mtrand.pyi

```diff
@@ -0,0 +1,132 @@
+import builtins
+from typing import overload, Optional, Sequence, Tuple, TypeVar, Union
+
+from numpy import ndarray
+
+_Size = Union[int, Tuple[int, ...]]
+_T = TypeVar('_T')
+
+class RandomState:
+    def __init__(self, state: Optional[Union[int, ndarray]] = ...) -> None: ...
+    def bytes(self, length: int) -> builtins.bytes: ...
+    def dirichlet(self, alpha: Sequence[float], size: Optional[_Size] = ...
+                  ) -> ndarray: ...
+
+    @overload
+    def choice(self, a: int, *,
+               replace: bool = ..., p: Optional[Sequence[float]] = ...
+               ) -> int: ...
+    @overload
+    def choice(self, a: Sequence[_T], *,
+               replace: bool = ..., p: Optional[Sequence[float]] = ...
+               ) -> _T: ...
+    @overload
+    def choice(self, a: Union[int, Sequence[_T]], size: _Size,
+               replace: bool = ..., p: Optional[Sequence[float]] = ...
+               ) -> ndarray: ...
+
+    @overload
+    def beta(self, a: float, b: float) -> float: ...
+    @overload
+    def beta(self, a: float, b: float, size: _Size) -> float: ...
+    @overload
+    def beta(self, a: ndarray, b: float) -> ndarray: ...
+    @overload
+    def beta(self, a: float, b: ndarray) -> ndarray: ...
+    @overload
+    def beta(self, a: ndarray, b: ndarray) -> ndarray: ...
```

Comment:
This method can also accept arbitrary "array-like" types which it coerces to ndarrays, e.g., lists of numbers.

Reply:
Hm... I'm not sure how to represent "array-like" types. A naive approach would be something like:

```python
_ArrayLike = Union[Sequence[_T], ndarray]
```

But nested sequences are also `_ArrayLike` and MyPy doesn't have support for recursive types (which we'd need for arbitrarily nested sequences).